HomeworkNine/CarArray.java

```diff
@@ -0,0 +1,83 @@
+package HomeworkEight;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class CarArray {
+
+    private ArrayList<Car> carList = new ArrayList<>();
+
+    public void add(Car car) {
+        carList.add(car);
+    }
+
+    public ArrayList<Car> getCarList() {
+        return carList;
+    }
+
+    public ArrayList<Car> sort() {
+        ArrayList<Car> sortedCars = carList;
+        sortedCars.sort(new YearComparator());
+        return sortedCars;
+    }
+
+    public ArrayList<Car> getRedCars(String color) {
+        ArrayList<Car> redCarsList = new ArrayList<>();
+
+        for (Car car : carList) {
+            if (car.getColor().equals(color)) {
+                redCarsList.add(car);
+            }
+        }
+        return redCarsList;
+    }
+
+    public HashMap<String, Integer> getColorStatistic() {
+        HashMap<String, Integer> hashMap = new HashMap<>();
+        for (int i = 0; i < carList.size(); i++) {
+            int count = 0;
+            if (hashMap.containsKey(carList.get(i).getColor())) {
+                continue;
+            }
+            for (Car car : carList) {
+                if (carList.get(i).getColor().equals(car.getColor())) {
+                    count++;
+                }
+            }
+            hashMap.put(carList.get(i).getColor(), count);
+        }
+        return hashMap;
+    }
+
+    public ArrayList<Car> getNewCars(int n) {
+        ArrayList<Car> newCars = new ArrayList<>();
```

Comment:
you don't have to copy list by objects, you can reassign list with filed's value. Just use code from sort() method:
ArrayList<Car> sortedCars = carList;